File: tests/fixtures/ldap_injection/c/baseline_constant_ldap.c
/* Baseline: filter is a string literal, no LDAP_INJECTION finding. */
#include <ldap.h>

int do_lookup(LDAP *ld) {
    LDAPMessage *res = NULL;
    return ldap_search_ext_s(
        ld,
        "ou=people,dc=example,dc=com",
        LDAP_SCOPE_SUBTREE,
        "(objectClass=person)",
        NULL, 0, NULL, NULL, NULL, 0, &res);
}

File: tests/fixtures/ldap_injection/go/baseline_constant_ldap.go
// Baseline: filter is a literal string, no taint reaches NewSearchRequest.
package ldap_baseline

import (
	"github.com/go-ldap/ldap/v3"
)

func Lookup() {
	conn, _ := ldap.DialURL("ldap://example.com")
	req := ldap.NewSearchRequest(
		"ou=people,dc=example,dc=com",
		ldap.ScopeWholeSubtree,
		ldap.NeverDerefAliases,
		0, 0, false,
		"(objectClass=person)",
		[]string{"cn"},
		nil,
	)
	conn.Search(req)
}
